tachys/src/html/style.rs

```diff
@@ -170,6 +170,80 @@ pub trait IntoStyle: Send {
     fn resolve(self) -> impl Future<Output = Self::AsyncOutput> + Send;
 }
 
+impl<T: IntoStyle> IntoStyle for Option<T> {
+    type AsyncOutput = Option<T::AsyncOutput>;
+    type State = (crate::renderer::types::Element, Option<T::State>);
+    type Cloneable = Option<T::Cloneable>;
+    type CloneableOwned = Option<T::CloneableOwned>;
+
+    fn to_html(self, style: &mut String) {
+        if let Some(t) = self {
+            t.to_html(style);
+        }
+    }
+
+    fn hydrate<const FROM_SERVER: bool>(
+        self,
+        el: &crate::renderer::types::Element,
+    ) -> Self::State {
+        if let Some(t) = self {
+            (el.clone(), Some(t.hydrate::<FROM_SERVER>(el)))
+        } else {
+            (el.clone(), None)
+        }
+    }
+
+    fn build(self, el: &crate::renderer::types::Element) -> Self::State {
+        if let Some(t) = self {
+            (el.clone(), Some(t.build(el)))
+        } else {
+            (el.clone(), None)
+        }
+    }
+
+    fn rebuild(self, state: &mut Self::State) {
+        let el = &state.0;
+        let prev_state = &mut state.1;
+        let maybe_next_t_state = match (prev_state, self) {
+            (Some(_prev_t_state), None) => {
+                Rndr::remove_attribute(el, "style");
```

Comment:
I don't think this logic quite works, because `IntoStyle` doesn't do the same thing for every type. For example, the implementation for the tuples (which is what the `style:color=foo` kind of syntax becomes) updates individual pieces of the `CSSStyleDeclaration`, rather than setting and unsetting the `style`. This `Option<T>` logic definitely works for the string-like types but not the tuple-like types.

Reply:
@gbj  If you're meaning here this `Option<impl IntoClass>` applies to e.g. `style:foo` it doesn't that's implemented as `impl<F, S> IntoStyle for (&'static str, F)` so would be `(&'static str, Option<F>)` for the optional case, something else not matched by this impl.

If you're meaning the case where e.g. `style=foo.then_some("color: red") style:background="transparent"` in the same tag, then yes there's a problem here as e.g. setting the `foo` to `None` would remove the `background` style too, but: 
1. I don't know if this is actually supported, both `style=""` and `style:attr` in the same tag.
2. I just tried solving by just changing the impl to using `T::default()` and calling the `T` trait methods directly, instead of the `None` handling and removing the attribute when `None` in this impl,  but this issue would still persist, so this would be a bug elsewhere in the renderer anyway if both versions in same tag is a supported patter.

For 2. you can see this happens here by just toggling the b prop from a parent component from true to false: 
```rust
#[component]
pub fn Sub(b: bool) -> impl IntoView {
    view! {
        <p style={ if b {"color: red" } else { "" }} style:background="transparent">"Sub"</p>
    }
}
```

Reply:
Here's an example of what I mean:
```rust
#[component]
pub fn App() -> impl IntoView {
    let (value, set_value) = signal(0);

    div()
        .style("font-weight: bold")
        .style(move || (value.get() % 2 == 0).then_some(("color", "red")))
        .on(ev::click, move |_| *set_value.write() += 1)
        .child("Click me")
}
```
When you click this, the `rebuild()` definition for the inner `Option<(&'static str, &'static str)>` will un-set the entire `style` attribute, not just the `color` property.

This can't be created using the view macro, but can be created pretty easily using the builder.

Reply:
No; this is not the same problem. The problem you describe in #3141 is that updating the entire `style` attribute overwrites individual styles, which is true. The problem I am describing is that updating individual styles can overwrite the rest of the `style` attribute, which should not be true.

Reply:
Although to be fair; the way I wrote the example is a little ambiguous as to whether you'd think it should update the whole style, or just that one style. I don't know. IMO this introduces a bug but maybe it would be so unusual for someone to actually create it that it's worth the trade off.

Reply:
I've provided a fix which seems to work for your example above, it starts going a bit further than my internal knowledge though.

Reply:
Effectively just adding the style attr name to the states of implementations using it, then implementing a reset of the css property.